src/lib/dev_util/proving.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  derDecode,
+  hexToBigInt,
+  publicKeyFromString,
+  getPublicInputsFromSignature,
+} from "babyjubjub-ecdsa";
+import { NonceSignature } from "../dev_mockData/nonceSigs";
+import { Jubmoji } from "../dev_types";
+import { recoverCounterMessageHash } from "./signature";
+import { getJubmojiPubKeyFromIndex } from "./utils";
+
+export const getJubmojiFromNonceSignature = ({
+  nonce,
+  rand,
+  sig,
+  pubKeyIndex,
+}: NonceSignature): Jubmoji => {
+  const decodedSig = derDecode(sig);
+  const msgHash = hexToBigInt(recoverCounterMessageHash(nonce, rand));
+  const pubKey = publicKeyFromString(getJubmojiPubKeyFromIndex(pubKeyIndex));
```

Comment:
Any reason the `getJubmojiPubKeyFromIndex` doesn't just return the public key itself instead of in string format?

Reply:
Oh I see is it to get Edwards form?

Reply:
Nope you're right, fixed